api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingBatchUpdateService.java

```diff
@@ -54,19 +63,42 @@ public void checkFreeTierBillingUsage() {
     long numberOfUsers = Iterators.size(freeTierActiveWorkspaceCreators.iterator());
     int count = 0;
 
+    Map<String, Double> allBQCosts = getFreeTierWorkspaceCostsFromBQ();
+
+    logger.info(String.format("Retrieved all BQ costs, size is: %d", allBQCosts.size()));
+
     for (List<DbUser> usersPartition :
         Iterables.partition(
             freeTierActiveWorkspaceCreators, freeTierCronUserBatchSizeFromConfig())) {
       logger.info(
           String.format(
               "Processing users batch of size/total: %d/%d. Current iteration is: %d",
               usersPartition.size(), numberOfUsers, count++));
-      freeTierBillingService.checkFreeTierBillingUsageForUsers(new HashSet<>(usersPartition));
+      freeTierBillingService.checkFreeTierBillingUsageForUsers(
+          new HashSet<>(usersPartition), allBQCosts);
     }
 
     logger.info("Checking Free Tier Billing usage - finish");
   }
 
+  private Map<String, Double> getFreeTierWorkspaceCostsFromBQ() {
+    final QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(
+                "SELECT id, SUM(cost) cost FROM `"
+                    + workbenchConfigProvider.get().billing.exportBigQueryTable
+                    + "` WHERE id IS NOT NULL "
+                    + "GROUP BY id ORDER BY cost desc;")
```

Comment:
Note: It is still `project.id` in the production. But I think we should switch to id. I don't know if there is any easier & backward compatibility way to migrate this query. 
One thing we can do: We could do work with Neil see next week after release. And ask him to switch query right after release?